File: 1582.Special-Positions-in-a-Binary-Matrix/Solution.py
class Solution:
    def numSpecial(self, mat: List[List[int]]) -> int:
        rows, cols = (len(mat), len(mat[0]))
        boolmat = [[True for _ in range(cols)] for _ in range(rows)]
             
        # Check rows
        for i in range(rows):
            if mat[i].count(1) != 1 or sum(mat[i]) != 1:
                boolmat[i] = [False] * cols

        # Check columns
        for j in range(cols):
            column = [mat[i][j] for i in range(rows)]
            if column.count(1) != 1 or sum(column) != 1:
                for row in boolmat:
                    row[j] = False

        # Count valid elements
        result = 0
        for i in range(rows):
            for j in range(cols):
                if mat[i][j] == 1 and boolmat[i][j]:
                    result += 1

        return result

File: 1582.Special-Positions-in-a-Binary-Matrix/Solution.java
class Solution {
    public int numSpecial(int[][] mat) {
        int rows = mat.length, cols = mat[0].length;
        
        int[] colSum = new int[cols];
        int[] rowSum = new int[rows];
        for(int row = 0; row < rows; row++){
            for(int col = 0; col < cols; col++){
                rowSum[row] += mat[row][col];
                colSum[col] += mat[row][col];
            }
        }
        boolean[] rowCountOne = new boolean[rows];
        boolean[] colCountOne = new boolean[cols];
        for(int i = 0; i < Math.max(rows, cols); i++){
            if(i < rows && rowSum[i] == 1){
                rowCountOne[i] = true;
            }
            if(i < cols && colSum[i] == 1){
                colCountOne[i] = true;
            }
        }
        int result = 0;
        for(int row = 0; row < rows; row++){
            for(int col = 0; col < cols; col++){
                if(rowCountOne[row] && colCountOne[col] && mat[row][col] == 1){
                    result++;
                }
            }
        }
        return result; 
    }
}
